packages/cloud_firestore/cloud_firestore/ios/Classes/FLTFirebaseFirestorePlugin.m

```diff
@@ -451,8 +451,8 @@ - (void)documentGet:(id)arguments withMethodCallResult:(FLTFirebaseMethodCallRes
 
 - (void)queryGet:(id)arguments withMethodCallResult:(FLTFirebaseMethodCallResult *)result {
   FIRQuery *query = arguments[@"query"];
-
-  if (query == nil) {
+  // Why we check [NSNull null]:  https://github.com/firebase/flutterfire/issues/9328
+  if (query == nil || query == [NSNull null]) {
```

Comment:
Should we change this everywhere or is it just for the query object?

Reply:
Guess we should

Reply:
Would it happen with Swift? Would this kind of bugs be an argument to start migrating slowly?

Reply:
Context: `nil` happens when the value does not exist or is undefined in Dart, e.g  a key on a map did not exist when sent over to ios. NSNull however is when a `null` value is explicitly sent from Dart, the method channel standard codec serializes it to NSNull explicitly instead of null

So swift all depends on what the message codec for a swift method channel does to convert it, my guess still some form of nsnull?

When I did most of the native modules for other FF packages I tried to avoid sending null values from dart, e.g if a value on a map is null then don't have that key in the map, that way only `nil` checking was required, I probably should have code commented this somewhere sorry

Reply:
Thanks for the explanation!

Reply:
Agree, lets not block this fix